lib/improver/tests/psychrometric_calculations/psychrometric_calculations/test_FallingSnowLevel.py

```diff
@@ -103,90 +103,124 @@ def setUp(self):
+                                [100.0, 5.0, 100.0],
+                                [100.0, 100.0, 100.0]])
+        self.land_sea = np.ones((3, 3))
+
+    def test_basic(self):
+        """Test it fills in the points it's meant to."""
         plugin = FallingSnowLevel()
-        orog = self.orog
-        orog[1, 1] = 0.0
-        land_sea = self.land_sea
-        land_sea[1, 1] = 0.0
-        highest_wb_int = self.highest_wb_int
-        highest_wb_int[1, 1] = 100.0
+        self.land_sea[1, 1] = 0
+
         expected = np.array([[1.0, 1.0, 2.0],
-                             [1.0, 300.0, 2.0],
+                             [1.0, 0, 2.0],
                              [1.0, 2.0, 2.0]])
-        result = plugin.fill_in_missing_data(self.snow_level_data,
-                                             orog, land_sea, highest_wb_int,
-                                             self.highest_height)
-        self.assertIsInstance(result, np.ndarray)
-        self.assertArrayEqual(result, expected)
+        plugin.fill_in_sea_points(self.snow_level_data, self.land_sea,
+                                  self.wb_int)
+        self.assertArrayEqual(self.snow_level_data, expected)
 
-    def test_nonfreezing_points(self):
-        """Test with integral above threshold sets snow level to highest_level
-            plus orograpy where the data can not be
-            interpolated from other points and points are not
-            sea-level points."""
+    def test_no_sea(self):
+        """Test it only fills in sea points, and ignores a land point"""
         plugin = FallingSnowLevel()
-        highest_wb_int = self.highest_wb_int * 100.0
-        expected = np.array([[301.0, 301.0, 301.0],
-                             [1.0, 1.5, 2.0],
-                             [1.0, 2.0, 301.0]])
-        result = plugin.fill_in_missing_data(self.snow_data_no_interp,
-                                             self.orog, self.land_sea,
-                                             highest_wb_int,
-                                             self.highest_height)
-        self.assertIsInstance(result, np.ndarray)
-        self.assertArrayEqual(result, expected)
+        expected = np.array([[1.0, 1.0, 2.0],
+                             [1.0, np.nan, 2.0],
+                             [1.0, 2.0, 2.0]])
+        plugin.fill_in_sea_points(self.snow_level_data, self.land_sea,
+                                  self.wb_int)
+        self.assertArrayEqual(self.snow_level_data, expected)
 
-    def test_freezing_points(self):
-        """Test with integral below threshold sets snow level to missing_value
-            where the data can not be interpolated from other points
-            and points are not a sea-level points."""
+    def test_all_above_threshold(self):
+        """Test it doesn't change points that are all above the threshold"""
         plugin = FallingSnowLevel()
-        expected = np.array([[-300.0, -300.0, -300.0],
+        self.wb_int[1, 1] = 100
```

Comment:
Does this test not need to turn point [1, 1] into a sea point, otherwise, it's really the same as the test above?

Reply:
Yes, I think you're right. I've changed this.